web-app/HeimScripts/heatmap/run.R

```diff
@@ -104,6 +106,30 @@ buildFields <- function(df){
   return(df)
 }
 
+writeDataForZip <- function(df, patientIDs) {
+  pidCols    <- as.character(patientIDs)
+  t          <- df
+  t[pidCols] <- lapply(t[pidCols], function(v) { (v - df$MEAN) / df$SD })
+  allCols    <- c(c("Row.Label", "Bio.marker", "MEAN", "SD", "SIGNIFICANCE"), pidCols)
+  write.table(t[allCols], "heatmap_data.tsv", sep="\t", na="", row.names=F, col.names=T)
+
+  allCols <- c(c("Row.Label"), pidCols)
+  write.table(df[allCols], "heatmap_orig_values.tsv", sep="\t", na="", row.names=F, col.names=T)
+}
+
+writeRunParams <- function(max_rows, sorting) {
+  params <- list(
+    max_rows = max_rows,
+    sorting  = sorting
+  )
+
+  if (exists("preprocessed") && exists("preprocessing_params")) {
+    params <- c(params, preprocessing_params)
+  }
+
+  write(toJSON(params, pretty=TRUE), 'params.json')
```

Comment:
Is json the most user friendly way of representing the parameters? Maybe a plain .txt (with one parameter per row) is easier?  @jannekevdp